src/net_processing.cpp

```diff
@@ -3844,6 +3851,12 @@ bool PeerLogicValidation::SendMessages(CNode* pto)
                     }
                 }
 
+                // cache the min fee rate for a txn to be included in a block
+                // applied as rebroadcast filter above
+                if (mempool.m_next_min_fee_cache < current_time){
+                    mempool.CacheMinRebroadcastFee();
```

Comment:
Did you benchmark `CacheMinRebroadcastFee` ? Not sure if it's the best way to encumber ThreadMessageHandler with mempool computation. I think it could be scheduled by the scheduler like we are doing for DumpBanList or DumpAddresses.